16_wordle/main_test.go

```diff
@@ -35,3 +36,29 @@ func TestContains(t *testing.T) {
 		}
 	}
 }
+
+func TestCheckCharPrecise(t *testing.T) {
+	tests := []struct {
+		input  []string
+		chosen []string
+		want   []int
+	}{
+		{input: []string{"a", "a", "a", "a", "a"},
+			chosen: []string{"a", "a", "a", "a", "b"},
+			want:   []int{2, 2, 2, 2, 1},
+		},
+		{
+			input:  []string{"a", "c", "a", "a", "b"},
+			chosen: []string{"a", "a", "a", "b", "a"},
+			want:   []int{2, 0, 2, 1, 1},
+		},
```

Comment:
Also, when you have converted into a loop, you can test with input with different length, so it becomes easier to test for special cases